table/block_based/block_based_table_reader.cc

```diff
@@ -2680,7 +2683,7 @@ void BlockBasedTable::MultiGet(const ReadOptions& read_options,
         PERF_COUNTER_BY_LEVEL_ADD(bloom_filter_full_true_positive, 1,
                                   rep_->level);
       }
-      if (s.ok()) {
+      if (s.ok() && !iiter->status().IsNotFound()) {
```

Comment:
```iiter``` may not necessarily have been initialized for this key and might have a stale status. The ```iiter``` status should only be considered if ```!first_block```.

Reply:
You're suggesting I try to fix a bug in which a failure status (not OK, not NotFound) could falsely be attached to too many keys in a MultiGet batch? That just seems like a case where "best effort" could be better, whereas this PR is about fixing an incorrect result. I'd prefer not to mix the two.